GitCommands/Git/Executable.cs

```diff
@@ -179,7 +203,7 @@ public StreamReader StandardError
                         throw new InvalidOperationException("Process was not created with redirected output.");
                     }
 
-                    return _process.StandardError;
+                    return _throwOnErrorOutput ? new StreamReader(new MemoryStream()) : _process.StandardError;
```

Comment:
Please verify we're not leaking memory here, I can't see where we're disposing the `StreamReader` created here. `_process` gets disposed, and it will dispose its `StandardError` reader, but the new will be left dangling.

Reply:
Done. But can't the GC do it? It is very unlikely that this code path is reached, since the exception already contains the consumed `StandardError` output.